File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    
    queue = [(0,0)]
    ans = 0
    
    while queue:
        summ, ind = queue.pop()
        if ind == len(numbers) and summ == target:
            ans += 1
            continue
        elif ind < len(numbers):
            now = numbers[ind]
            queue.append((summ+now, ind+1))
            queue.append((summ-now, ind+1))
    
    return ans

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int ans;
    static int targetInd;
    static int targetNum;
    static int[] n_lis;
    
    public int solution(int[] numbers, int target) {
        targetInd = numbers.length;
        targetNum = target;
        n_lis = numbers;
        ans = 0;
        recur(0,0);
        
        return ans;
    }
    
    public void recur(int ind, int sum){
        if(ind == targetInd){
           if(sum == targetNum){
              ans += 1;
           }
        }
        else{
            recur(ind+1, sum-n_lis[ind]);
            recur(ind+1, sum+n_lis[ind]);
        }
    }
}